ddl/ddl_worker.go

```diff
@@ -149,6 +146,15 @@ func asyncNotify(ch chan struct{}) {
 // buildJobDependence sets the curjob's dependency-ID.
 // The dependency-job's ID must less than the current job's ID, and we need the largest one in the list.
 func buildJobDependence(t *meta.Meta, curJob *model.Job) error {
+	switch curJob.Type {
+	case model.ActionAddIndex:
+		t.SetJobListKey(meta.DefaultJobListKey)
+		defer t.SetJobListKey(meta.AddIndexJobListKey)
+	default:
+		t.SetJobListKey(meta.AddIndexJobListKey)
+		defer t.SetJobListKey(meta.DefaultJobListKey)
+	}
+
 	jobs, err := t.GetAllDDLJobs()
```

Comment:
I suggest rename `GetAllDDLJobs` to `GetDDLJobsInQueue` and pass the queue, rather than change meta's `jobListKey` status, it's very tricky.

Reply:
GetAllDDLJobs definitely get jobs from `mDDLJobListKey`? Is that correct?

Reply:
`mDDLJobListKey` will change between two queue, I don't  know   which one,  that's  the  problem.

So I suggest:
```
GetDDLJobsInQueue(general)
GetDDLJobsInQueue(addindex)
```

Reply:
Please address comment

Reply:
Done, I changed the name of the function.